src/AccessibilityInsights.Extensions.AzureDevOpsTests/AzureBoardsIssueReportingTests.cs

```diff
@@ -0,0 +1,268 @@
+        }
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void TruncateSelectedFields_RuleSourceIsSpecified_AddsRuleSourceValue()
+        {
+            const string ruleSourceContent = "weather: cold and clear";
+
+            string originalRuleSource = '[' + ruleSourceContent + ']';
+            Dictionary<IssueField, string> issueFieldPairs = new Dictionary<IssueField, string>();
+            IssueInformation issueInfo = new IssueInformation(ruleSource: originalRuleSource);
+
+            PrivateType privateFileIssueHelpers = new PrivateType(typeof(FileIssueHelpers));
+            privateFileIssueHelpers.InvokeStatic("TruncateSelectedFields", issueInfo, issueFieldPairs);
+
+            Assert.AreEqual(4, issueFieldPairs.Count);
+            string modifiedRuleSource = issueFieldPairs[IssueField.RuleSource];
+            Assert.AreEqual(ruleSourceContent, modifiedRuleSource);
+            Assert.IsNull(issueFieldPairs[IssueField.ProcessName]);
+            Assert.IsNull(issueFieldPairs[IssueField.Glimpse]);
+            Assert.IsNull(issueFieldPairs[IssueField.TestMessages]);
+        }
+
+#if FAKES_SUPPORTED
+        [TestMethod]
+        [Timeout(1000)]
+        public void CreateIssuePreviewAsync_TeamNameIsNull_ChainsThroughCorrectly()
+        {
+            using (ShimsContext.Create())
+            {
+                const string expectedProjectName = "Super Project";
+                const string expectedTeamName = null;
+                string actualProjectName = null;
+                string actualTeamName = null;
+                Uri expectedUri = new Uri("https://www.bing.com");
+
+                AzureDevOpsIntegration integration = new ShimAzureDevOpsIntegration
+                {
+                    CreateIssuePreviewStringStringIReadOnlyDictionaryOfAzureDevOpsFieldString = (p, t, f) =>
+                    {
+                        actualProjectName = p;
+                        actualTeamName = t;
+                        return expectedUri;
+                    },
+                    ConnectedToAzureDevOpsGet = () => true,
+                };
+
+                ShimAzureDevOpsIntegration.GetCurrentInstance = () => integration;
+
+                ConnectionInfo connectionInfo = new ConnectionInfo(expectedUri,
+                    new TeamProject(expectedProjectName, Guid.Empty),
+                    new Team(expectedTeamName, Guid.Empty));
+                IssueInformation issueInfo = new IssueInformation();
+
+                Uri actualUri = FileIssueHelpers.CreateIssuePreviewAsync(connectionInfo, issueInfo).Result;
+
+                Assert.AreEqual(expectedUri, actualUri);
+                Assert.AreEqual(expectedProjectName, actualProjectName);
+                Assert.IsNull(actualTeamName);
+            }
+        }
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void CreateIssuePreviewAsync_TeamNameIsNotNull_ChainsThroughCorrectly()
```

Comment:
This test fails 100% of the time on my machine

Reply:
Tests should work now. Not sure about the Run all issue...

Reply:
Unit tests pass. I've pushed a commit that unblocks the unit tests for me